### PR TITLE
Domains: more robust form

### PR DESCRIPTION
**Pull request by stsewd**
This form was even accepting spaces...

Based on the regex from https://api.cloudflare.com/
(`pattern: ^([a-zA-Z0-9][\-a-zA-Z0-9]*\.)+[\-a-zA-Z0-9]{2,20}$`)

Closes https://github.com/readthedocs/readthedocs.org/issues/1707

### CODE REVIEW

**Comment by humitos** [comment]
> Based on the regex from [api.cloudflare.com](https://api.cloudflare.com/)
> (`pattern: ^([a-zA-Z0-9][\-a-zA-Z0-9]*\.)+[\-a-zA-Z0-9]{2,20}$`)

This should probably a comment in the code next to the regex. With this link, that goes directly to the pattern: https://api.cloudflare.com/#zone-create-zone

File: readthedocs/projects/validators.py [hunk 1]
```diff
@@ -12,39 +10,13 @@
...
-        except ValidationError as exc:
-            if not self.accept_idna:
-                raise
-            if not value:
-                raise
-            try:
-                idnavalue = value.encode('idna')
-            except UnicodeError:
-                raise exc
-            super().__call__(idnavalue)
```
Comment (agjohnson) [inline readthedocs/projects/validators.py:47]: What's up with this removal? I don't really even understand this code, was the `accept_idna` attribute used at all?

**Review by agjohnson — APPROVED** [review]
Looks great. Noted what looks to be a removal of unused code. It might be worth double checking we weren't using the attribute on commercial/etc. Based on this PR it looks like the removed code was ancient and wasn't even used.

File: readthedocs/projects/validators.py (hunk 1 above)
Comment (stsewd) [reply readthedocs/projects/validators.py:47]: Nope, this was just copy/pasta from the original PR in django itself, we aren't using it.

Comment (stsewd) [reply readthedocs/projects/validators.py:47]: I just double checked corporate and ext, we are good

Comment (agjohnson) [reply readthedocs/projects/validators.py:47]: Forgot to ping you on this one @stsewd :point_up:

Comment (stsewd) [reply readthedocs/projects/validators.py:47]: Already replied if that's what you mean p: maybe github didn't update for you?

Comment (agjohnson) [reply readthedocs/projects/validators.py:47]: oops! yeah, github didn't update for me

**Pull request merged**